src/darker/git.py

```diff
@@ -149,23 +153,32 @@ def _git_check_output_lines(
     """Log command line, run Git, split stdout to lines, exit with 123 on error"""
     logger.debug("[%s]$ %s", cwd, " ".join(cmd))
     try:
-        return check_output(
+        result = run(
             ["git"] + cmd,
             cwd=str(cwd),
+            check=True,
             encoding="utf-8",
+            stdout=PIPE,
             stderr=PIPE,
             env={"LC_ALL": "C"},
-        ).splitlines()
+        )
+        return result.stdout.splitlines()
     except CalledProcessError as exc_info:
+        retval = exc_info.returncode
+        msg = exc_info.stderr
+        if (retval == 129 and msg.startswith("Not a git repository")) or (
+            retval == 1 and msg.startswith("error: could not access ")
+        ):
+            raise NotGitRespository(f"{cwd} is not a git repository") from exc_info
```

Comment:
I wonder if this is a good way to detect non-Git-directories. The `could not access 'HEAD'` message came up when tests were run on GitHub, but I can't reproduce it locally. We don't know whether that error might occur within Git repositories under different conditions, and this might mask such situations.

Reply:
...or could there be an alternative explanation why Git would terminate with `error: Could not access 'HEAD'` than a different Git version?

Pytest temporary directories being created in a directory tree which is a Git repository? Sounds unlikely to me.

Reply:
No, this is indeed most probably a difference in Git.

In a Debian 11 Docker image:
```
# git --version
git version 2.30.2
# git diff --name-only --relative HEAD -- .
error: Could not access 'HEAD'
```

Same on Debian Buster:
```
# git --version
git version 2.20.1
root@6c76cb727ad0:/# git diff --name-only --relative HEAD -- .
Not a git repository
To compare two paths outside a working tree:
usage: git diff [--no-index] <path> <path>
```

Same on Fedora 30
```
# git --version
git version 2.21.3
# git diff --name-only --relative HEAD -- .
Not a git repository
To compare two paths outside a working tree:
usage: git diff [--no-index] <path> <path>
```

Reply:
For a non-existing revision, I get this on Git 2.30.2:
```
# git diff --name-only --relative foo -- .
fatal: bad revision 'foo'
```

So at least that won't be confused with a non-Git-directory.

Reply:
Maybe this would be a more reliable detection:

Git 2.21.3 and 2.30.2:
```
# git rev-parse --is-inside-work-tree
true
```

Git 2.21.3:
```
# git rev-parse --is-inside-work-tree
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

Git 2.30.2:
```
# git rev-parse --is-inside-work-tree
fatal: not a git repository (or any of the parent directories): .git
```

Reply:
Oops, there really is something funky going on in the GitHub Actions environment!

In `test_main_in_plain_directory(tmp_path, ...)`, when `main()` calls `git_is_repository()`, it gets `True` as the result even though we should be in a clean temporary directory!

See [this build](https://github.com/akaihola/darker/pull/140/checks?check_run_id=3645005237) for details.

Reply:
**HEISENBUG!**

I added this test in `test_git.py`:
```python
def test_tmp_path_sanity(tmp_path):
    """Make sure Pytest temporary directories aren't inside a Git repository"""
    try:
        result = git._git_check_output_lines(
            ["rev-parse", "--absolute-git-dir"], tmp_path, exit_on_error=False
        )
    except CalledProcessError as exc_info:
        if exc_info.returncode != 128 or not exc_info.stderr.startswith(
            "fatal: not a git repository"
        ):
            raise
    else:
        output = "\n".join(result)
        raise AssertionError(
            f"Temporary directory {tmp_path} for tests is not clean."
            f" There is a Git directory in {output}"
        )
```

And now all tests pass! How did this change temporary directories in other tests?!?